crates/matrix-sdk-sled/src/state_store.rs

```diff
@@ -299,24 +313,49 @@ impl SledStore {
         &self,
         event: &[u8],
     ) -> Result<T, SledStoreError> {
-        if let Some(key) = &*self.store_key {
-            let encrypted: EncryptedEvent = serde_json::from_slice(event)?;
-            Ok(key.decrypt(encrypted)?)
+        if let Some(key) = &*self.store_cipher {
+            Ok(key.decrypt_value(event)?)
         } else {
             Ok(serde_json::from_slice(event)?)
         }
     }
 
-    pub async fn save_filter(&self, filter_name: &str, filter_id: &str) -> Result<()> {
-        self.session.insert(("filter", filter_name).encode(), filter_id)?;
+    fn encode_key<T: EncodeSecureKey + EncodeKey + ?Sized>(
+        &self,
+        table_name: &str,
+        key: &T,
```

Comment:
Since `&impl EncodeKey` implements `EncodeKey`, this is strictly more general when you change this line to `key: T`, remove `Sized` above and just pass references where it makes sense (in places where you pass a tuple you can then remove the extra `&` in front).